.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,65 @@
+name: Shellcheck Lint
+
+on:
+  push:
+    paths:
+      # Run workflow on every push
+      # only if a file within the specified paths has been changed:
+      - 'tests/**'
+      - 'usr/**'
+
+  pull_request:
+    paths:
+      # Run workflow on every push
+      # only if a file within the specified paths has been changed:
+      - 'tests/**'
+      - 'usr/**'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Shellcheck Lint
+
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+
+    steps:
+      # Required to access files of this repository
+      - uses: actions/checkout@v2
+
+      # Download Shellcheck and add it to the workflow path
+      - name: Download Shellcheck
+        run: |
+          wget -qO- "https://github.com/koalaman/shellcheck/releases/download/stable/shellcheck-stable.linux.x86_64.tar.xz" | tar -xJv
+          chmod +x shellcheck-stable/shellcheck
+      # Verify that Shellcheck can be executed
+      - name: Check Shellcheck Version
+        run: |
+          shellcheck-stable/shellcheck --version
+
+      # Run Shellcheck on repository
+      # ---
+      # https://github.com/koalaman/shellcheck
+      # ---
+      # Excluded checks:
+      # https://www.shellcheck.net/wiki/SC1091 -- Not following: /etc/rc.status was...
+      # https://www.shellcheck.net/wiki/SC1090 -- Can't follow non-constant source. ..
+      # ---
+      - name: Run Shellcheck
+        run: |
+          set +e
+          find  ./*/ -type f | sort | while read -r sh; do
+            if [ "$(file --brief --mime-type "$sh")" == 'text/x-shellscript' ]; then
+              echo "shellcheck'ing $sh"
```

Comment:
`shellcheck` takes multiple files as an argument. You could therefore run it with a list of files instead of calling it once for each file. This could also speed up the process.

Reply:
The argument list might be toooooo long with so many rear-scripts to be shellcheck'ed. And I like the info on which script shellcheck is/was working ❤️  And we are checking for mime-type  == 'text/x-shellscript' to get everything that ist a shellscript right now. shellcheck'ing "*" won't be a good idea because there might be/are some files that are not a shellscript and will also get shellcheck'ed with false-errors. For the beginning I propose to start with this and optimize later. Agree?

Running the part from GH action locally with only 32 "real" shellscripts:
```
real	0m43.018s
user	0m6.095s
sys	0m2.084s
```
Running the GH action with [act](https://github.com/nektos/act) locally with only 32 "real" shellscripts:
```
real	0m14.570s
user	0m0.115s
sys	0m0.058s
```

Vs. `shellcheck $(find  ./*/ -type f | sort)` locally with 955 files (regardless if they're shellscripts or not):
```
real	0m19.349s
user	0m0.069s
sys	0m0.086s
```